### DISCUSSION TITLE
How to get or display / print the overall build time of an cmake project?

### DISCUSSION THREAD

**Discussion by developsessions** [microsoft/vscode-cmake-tools#3003] [Q&A] [answered]
How can I display the build time of an cmake project in vs code? It's not possible to wrap the cmake.exe command right?
I am a Windows user.

**Comment by elahehrashedi** [comment]
One way is to run `"CMake: Show Build Command"` to get the command string, for example, this is a build command: `"C:\Program Files\CMake\3.25.2\bin\cmake.EXE" --build d:/tests/cmakepreset/build --config Debug --target all` , and then use some methods to get the execution time of this command. You can find some examples for measuring the execution time of a command on windows from here: [link ](https://stackoverflow.com/questions/673523/how-do-i-measure-execution-time-of-a-command-on-the-windows-command-line)

**Reply by bobbrow** [reply]
Another option is to use one of the "Visual Studio" generators.  When you build with those, MSBuild will print out the elapsed time at the end of the build.

We actually already capture the build time in the extension too.  It would be pretty easy for us to just print it out.

**Reply by developsessions** [reply]
@elahehrashedi I know but then I have to execute the command seperately in an command line and cannot use the build button of the cmake tools extension.

@bobbrow Another generator is not possible because I need that makefiles for an very special cross compiler.

If its easy to print out the time this would be an amazing feature, beacause every developer who write cmake files they are also interessted to decrease and optimize the compile time.

Another nice feature would be to have that resulting build time in the bottom status bar.

**Reply by bobbrow** [reply]
The extension will print out the build time starting in Monday's pre-release.
We won't be adding it to the status bar because we're in the process of removing our buttons, etc from the status bar and putting them in the side bar.

**Reply by developsessions** [reply]
@bobbrow Amazing. Thank you. I will have a look at it.

**Comment by elahehrashedi** [comment, marked as answer]
@developsessions can you open a new issue and request this feature there?

**Reply by bobbrow** [reply]
There is a link to create an issue from a discussion that I clicked just now. #3008
